src/cljs/athens/keybindings.cljs

```diff
@@ -183,89 +202,81 @@
+   ["Yesterday"     mui-icons/Today #(str "[[" (:title (get-day 1)) "]]") nil]
+   ["YouTube Embed" mui-icons/YouTube "{{[[youtube]]: }}" nil]
+   ["iframe Embed"  mui-icons/DesktopWindows "{{iframe: }}" nil]])
 
 ;;[mui-icons/ "Block Embed" #(str "[[" (:title (get-day 1)) "]]")]
 ;;[mui-icons/DateRange "Date Picker"]
 ;;[mui-icons/Attachment "Upload Image or File"]
 ;;[mui-icons/ExposurePlus1 "Word Count"]
 
-
-;; TODO: also replace typeahead characters that follow "/". may need event to find selectionStart
-(defn select-slash-cmd
+(defn auto-complete-slash
   [index state]
   (let [{:keys [atom-string]} @state
-        [_ _ expansion _] (slash-options index)
+        [_ _ expansion _] (get slash-options index)
         expand (if (fn? expansion) (expansion) expansion)
         replace-str (subs atom-string 0 (dec (count atom-string)))
         new-str     (str replace-str expand)]
-    (swap! state merge {:search/index 0
-                        :search/type nil
-                        :generated-str  new-str})))
+    (swap! state assoc
+           :search/index 0
+           :search/type nil
+           :generated-str new-str)))
 
 
-(defn auto-complete
+(defn auto-complete-inline
   [state e completed-str]
   (let [{:keys [start head tail target]} (destruct-event e)
         {:search/keys [query type]} @state
-        head-pattern (cond
-                       (= type :block) (re-pattern (str "(.*)\\(\\(" query))
-                       (= type :page)  (re-pattern (str "(.*)\\[\\[" query)))
-        tail-pattern (cond
-                       (= type :block) #"(\)\))?(.*)"
-                       (= type :page)  #"(\]\])?(.*)")
-        new-head (cond
-                   (= type :block) "$1(("
-                   (= type :page)  "$1[[")
-        closing-str (cond
-                      (= type :block) "))"
-                      (= type :page)  "]]")
+        block? (= type :block)
+        page? (= type :page)
+        head-pattern (cond block? (re-pattern (str "(.*)\\(\\(" query))
+                           page?  (re-pattern (str "(.*)\\[\\[" query)))
+        tail-pattern (cond block? #"(\)\))?(.*)"
+                           page?  #"(\]\])?(.*)")
+        new-head (cond block? "$1(("
+                       page?  "$1[[")
+        closing-str (cond block? "))"
+                          page?  "]]")
         new-str (replace-first head head-pattern (str new-head completed-str closing-str))
         [_ closing-delimiter after-closing-str] (re-matches tail-pattern tail)]
-    (swap! state merge {:atom-string (str new-str after-closing-str)
-                        :search/query nil
-                        :search/type nil})
-    (when closing-delimiter (set! (. target -selectionStart) (+ 2 start)))))
+    (swap! state assoc
+           :generated-str (str new-str after-closing-str)
+           :search/index 0
+           :search/results nil
+           :search/query nil
+           :search/type nil)
```

Comment:
Do we need to set all of these to `nil`? Or is it good enough to just `:search/type` to `nil`. The next time dropdown is opened and `write-char` happens, these other values are set to correct initial values. It's okay for the other `:search/` values to be arbitrary as long as `:search/type` is nil